src/components/header/_macro.njk

```diff
@@ -169,7 +169,7 @@
         </div>
         <div class="ons-header__main">
             <div class="ons-container{{ ' ons-container--full-width' if params.fullWidth }}{{ ' ons-container--wide' if params.wide }}">
-                <div class="ons-grid ons-grid--gutterless ons-grid--flex ons-grid--between ons-grid--vertical-center ons-grid--no-wrap">
+                <div class="ons-grid {{ 'ons-grid--gutterless' if not params.button }} ons-grid--flex ons-grid--between ons-grid--vertical-center ons-grid--no-wrap">
```

Comment:
```suggestion
                <div class="ons-grid{{ ' ' + 'ons-grid--gutterless' if not params.button }} ons-grid--flex ons-grid--between ons-grid--vertical-center ons-grid--no-wrap">
```
Theres a space before and after this so if the class isn't set then it results in a double space in the class list. This syntax will stop this